File: server/src/com/broll/gainea/server/core/cards/events/E_GetCards.java
package com.broll.gainea.server.core.cards.events;

import com.broll.gainea.server.core.cards.EventCard;

public class E_GetCards extends EventCard {
    public E_GetCards() {
        super(35, "Markttag", "Jeder Spieler erhält eine Karte");
    }

    @Override
    protected void play() {
        game.getPlayers().forEach(player -> {
            player.getCardHandler().receiveCard(game.getCardStorage().getRandomPlayableCard());
        });
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/events/E_GetCards.kt
package com.broll.gainea.server.core.cards.events

import com.broll.gainea.server.core.cards.EventCard
import com.broll.gainea.server.core.utils.iteratePlayers


class E_GetCards : EventCard(35, "Markttag", "Jeder Spieler erhält eine Karte") {
    override fun play() {
        game.iteratePlayers(500) {
            it.cardHandler.receiveCard(game.cardStorage.getRandomPlayableCard())
        }
    }
}
